app/models/author.rb

```diff
@@ -81,16 +71,7 @@ def build_or_update(publication, contribution_hash = {})
   def contributions_changed_callback(*_args)
   end
 
-  has_many :publications, through: :contributions do
-    def approved
-      where("contributions.status='approved'")
-    end
-
-    def with_sciencewire_id
-      where(Publication.arel_table[:sciencewire_id].not_eq(nil))
-    end
-  end
-
```

Comment:
I can't approve this change without doing a git grep on all the possible uses of this.  Given that it's not broken and this entails more work for me to review it, why should it be done in this sprint?  Given my general lack of confidence in this app and it's specs at this time, I'm reluctant to change things that are not broken and spend the time on fixing an app that might be entirely replaced by the RIALTO project sprints.

Reply:
I already did that analysis, so I don't understand why it would be prohibitive for you.  There is no way I want to rely on anti-pattern methods defined dynamically per object in AR callbacks if we can avoid it.

Reply:
The output of `git grep with_sciencewire_id` is empty, if that what you are looking for.